### PR TITLE
Documentation and Examples

### PR DESCRIPTION
**Pull request by justusschock**
This PR adds documentation and examples for the new backends (and updates old examples). 

This PR should be merged after #102 and #172 were merged

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -3,9 +3,9 @@
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
 
-===================================
-delira - Deep Learning In RAdiology
-===================================
+=====================================================================
+delira - delira - A Backend Agnostic High Level Deep Learning Library
```
Comment (mibaumgartner) [inline docs/index.rst]: Only one delira :)

File: notebooks/classification_examples/pytorch.ipynb [hunk 2]
```diff
@@ -200,9 +163,131 @@
...
+    "class Flatten(torch.nn.Module):\n",
+    "        \n",
+    "    def forward(self, x):\n",
+    "        return x.view(x.size(0), -1)\n",
+    "\n",
+    "class SmallVGGPyTorch(AbstractPyTorchNetwork):\n",
+    "    def __init__(self, in_channels, num_classes):\n",
+    "        super().__init__()\n",
+    "        \n",
+    "        self.model = torch.nn.Sequential(\n",
```
Comment (mibaumgartner) [inline notebooks/classification_examples/pytorch.ipynb]: Maybe we should use a simpler model so it is a bit easier for beginners?

**Review by mibaumgartner — APPROVED** [review]
Only minor remarks, rest looks good :D

File: notebooks/classification_examples/pytorch.ipynb (hunk 2 above)
Comment (justusschock) [reply notebooks/classification_examples/pytorch.ipynb]: That's why I opted for VGG and not for any Resnet like model. What do you suggest?

Comment (mibaumgartner) [reply notebooks/classification_examples/pytorch.ipynb]: To demonstrate the general capability of training a NN with Delira we would only need a very small net. Some conv+pool and fc (around 3-4 layers) should be sufficient. Something like this https://www.tensorflow.org/tutorials/keras/basic_classification :)

**Comment by mibaumgartner** [comment]
Furthermore, we should also add the prediction of a datamanger with `experiment.test()` to the notebooks. Maybe even something with `experiment.resume()` .

File: notebooks/classification_examples/pytorch.ipynb (hunk 2 above)
Comment (justusschock) [reply notebooks/classification_examples/pytorch.ipynb]: Okay, I'll change this

**Comment by justusschock** [comment]
I don't know if this wouldn't be too much for the notebooks, we should maybe leave these notebooks as they are (after simplifying the model structure) and create another one containing a whole pipeline (with only one backend though).

**Comment by mibaumgartner** [comment]
Ok, we can do that, too :) 
Will you add the new notebook to this PR?

**Comment by justusschock** [comment]
Probably not, I will only simply the models in the existing notebooks and then this should be good to go. Otherwise this PR becomes way too huge

**Comment by justusschock** [comment]
@mibaumgartner is the architecture now simple enough?

File: notebooks/classification_examples/chainer.ipynb [hunk 3]
```diff
@@ -166,8 +166,8 @@
     "from batchgenerators.transforms.sample_normalization_transforms import MeanStdNormalizationTransform\n",
     "\n",
     "transforms = Compose([\n",
-    "    RandomCropTransform(200), # Perform Random Crops of Size 200 x 200 pixels\n",
-    "    ResizeTransform(224), # Resample these crops back to 224 x 224 pixels\n",
+    "    RandomCropTransform(24), # Perform Random Crops of Size 200 x 200 pixels\n",
```
Comment (mibaumgartner) [inline notebooks/classification_examples/chainer.ipynb]: Please adjust the comments accordingly.

File: notebooks/classification_examples/pytorch.ipynb [hunk 4]
```diff
@@ -118,8 +118,8 @@
     "from batchgenerators.transforms.sample_normalization_transforms import MeanStdNormalizationTransform\n",
     "\n",
     "transforms = Compose([\n",
-    "    RandomCropTransform(200), # Perform Random Crops of Size 200 x 200 pixels\n",
-    "    ResizeTransform(224), # Resample these crops back to 224 x 224 pixels\n",
+    "    RandomCropTransform(24), # Perform Random Crops of Size 200 x 200 pixels\n",
+    "    ResizeTransform(32), # Resample these crops back to 224 x 224 pixels\n",
```
Comment (mibaumgartner) [inline notebooks/classification_examples/pytorch.ipynb]: Please adjust the comments accordingly.

**Comment by justusschock** [comment]
So, after I did this, it is ready to merge? Or do you want additional simplifications?

**Comment by mibaumgartner** [comment]
Should be ready to merge, rest looks good and simple :D

**Pull request merged**